### PR TITLE
[#1653] Fix GraphQL overfetching issue

### PR DESCRIPTION
**Pull request by david-kubecka**
Make sure that EntityViewSetting.fetches are set up correctly,
i.e. they do not include intermediate/non-leaf fields as those
cause the whole subtree to be fetched from the DB.

### CODE REVIEW

File: integration/graphql/pom.xml [hunk 1]
```diff
@@ -59,6 +59,11 @@
             <artifactId>junit</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>3.6.28</version>
+        </dependency>
```
Comment (beikov) [inline integration/graphql/pom.xml:66]: Please use `<scope>test</scope>`

**Review by beikov — COMMENTED** [review]
Also, please add the issue number to the commit messages e.g. `[#1662] Fix GraphQL overfetching issue` and add a bug fix entry to the CHANGELOG.

File: integration/graphql/pom.xml (hunk 1 above)
Comment (david-kubecka) [reply integration/graphql/pom.xml]: done

**Comment by david-kubecka** [comment]
> Also, please add the issue number to the commit messages e.g. `[#1662] Fix GraphQL overfetching issue` and add a bug fix entry to the CHANGELOG.

done

**Review by beikov — APPROVED** [review]
Changes look good, but please squash the commits to one.

**Comment by david-kubecka** [comment]
@beikov In the end I had to modify the fix a little bit because it didn't work for custom/additional fields. Please have a look. Will you please run the CI workflow just for sure?

Also, I can surely squash the commits but why? :-) Don't you find the split into testing/refactoring/fixing commits easier to follow?

**Comment by beikov** [comment]
>Also, I can surely squash the commits but why? :-) Don't you find the split into testing/refactoring/fixing commits easier to follow?

I prefer having one commit per issue :)
When you squash things, can you please make sure the license headers are up-to-date?

File: examples/spring-data-graphql/pom.xml [hunk 2]
```diff
@@ -85,14 +85,21 @@
             <version>${project.version}</version>
         </dependency>
         <dependency>
-            <groupId>com.graphql-java</groupId>
+            <groupId>com.graphql-java-kickstart</groupId>
```
Comment (beikov) [inline examples/spring-data-graphql/pom.xml:88]: Looks like this change doesn't work for Spring Data 1.11.x:

```
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean found for dependency [graphql.kickstart.servlet.context.GraphQLServletContextBuilder]: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=false)}
```

Comment (david-kubecka) [reply examples/spring-data-graphql/pom.xml:88]: Yeah, the newer versions included in the upgraded `graphql-java-kickstart` no longer works with boot 1.11.*. I suggest dropping the profile (or faking it into 2.0 as is already done in https://github.com/Blazebit/blaze-persistence/blob/020ebd73b188613d824bdda8f18172365013a062/examples/spring-data-dgs/pom.xml#L206).

Comment (beikov) [reply examples/spring-data-graphql/pom.xml:88]: Faking it sounds fine to me, can you please make the changes and then squash everything? :)

Comment (david-kubecka) [reply examples/spring-data-graphql/pom.xml:88]: I realized that dropping the profile would still lead to running the test, although with the default profile version (2.7 in this case). So it really doesn't matter and I faked it into 2.0.

**Comment by beikov** [comment]
Great work, thanks!

**Pull request merged**